Large variant of Media object pattern

bartaz opened the pull request:
## Done

As part of implementing Snap details page we needed bigger variant of media pattern (as mentioned in #1315) that was implemented in https://github.com/canonical-websites/snapcraft-flask/pull/49

Here it is ported to base Vanilla.

It adds new modifier to `.p-media-object` class: `.p-media-object--large`. It has:
- that is a larger version of image 96x96px
- bigger title (defaults to heading 1 styles)
- larger spacing between paragraphs

## QA

- Pull code
- Run `./run serve --watch`
- Open http://0.0.0.0:8101/vanilla-framework/
- Go to 'Media object - large' example: http://0.0.0.0:8101/vanilla-framework/examples/patterns/media-object/media-object-large/

## Details

Fixes  #1315

## Screenshots

<img width="510" alt="screen shot 2017-09-14 at 15 06 47" src="https://user-images.githubusercontent.com/83575/30431598-5ad55462-995f-11e7-8343-84fe4a0d7547.png">

webteam-app commented:
Starting demo at: http://vanilla-framework-pr-1343.run.demo.haus/vanilla-framework/

barrymcgee commented on scss/_patterns_media-object.scss:
```diff
@@ -12,6 +12,13 @@
+  %media-image {
+    border-radius: 4px;
+    height: 3.75rem;
+    margin-right: 1rem;
```
Can you use [spacing variables](https://github.com/vanilla-framework/vanilla-framework/blob/develop/scss/_settings_spacing.scss) here, please?

anthonydillon commented:
As this is a block or pattern level modification. I think the modification should just be on the block and the elements within should respond accordingly.

bartaz commented:
@anthonydillon @barrymcgee Thanks for comments.

I updated PR to use spacing variables and moved modifier to block level class name `.p-media-object--large`.

bartaz commented:
@anthonydillon Sure. I just updated media object docs to include example of new variant.

webteam-app commented:
Starting demo at: http://vanilla-framework-pr-1343.run.demo.haus/vanilla-framework/

kwm14 commented:
## Visual review

Referencing @spencerbygraves design [spec](https://user-images.githubusercontent.com/9105999/29959607-0b1fd8bc-8ef0-11e7-90a8-cf3e17e956a4.png)

- [ ] Update element margin to: `2.5rem`
- [ ] Update content margin-top to: `0.75rem`

webteam-app commented:
Starting demo at: http://vanilla-framework-pr-1343.run.demo.haus/vanilla-framework/

kwm14 commented:
LGTM 👍

bartaz commented:
@kwm14 @barrymcgee 
I'm not sure if including top/bottom padding was intended for this pattern.

Yes, on snap details page media object has padding around it, but also it has bottom border - which makes it a good candidate to be used inside of `p-strip is-bordered` - and that's how it was implemented.

Now, with additional padding the spacing is doubled (both from media-object and strip patterns).
Removing strip will make bottom border disappear. Unless there is a possibility of making p-strip with border but without padding?

kwm14 commented:
@bartaz when reviewing your PR I synced up with @spencerbygraves and referenced his [spec](https://user-images.githubusercontent.com/9105999/29959607-0b1fd8bc-8ef0-11e7-90a8-cf3e17e956a4.png), so I was just following his guidelines when adding my design +1 to make sure the pattern matched what he had designed.

If this was a particular use-case to include top/bottom padding for this component I'm happy for you to remove and roll back to your build which had `1rem` padding top and bottom?

bartaz commented:
@kwm14 Thanks for comment.

I believe that the spec was meant to define spacing for snap details page, which in here we mistakenly interpreted as defining element margins for the pattern itself.

Large media pattern on it's own can keep default spacing defined by Vanilla and this will allow to use it inside of p-strip without adding unexpected spacing. I'll propose a bugfix for that.

kwm14 commented:
@bartaz ah ok thanks for clarifying that and thanks for filing a bug. I'll keep an eye on it as we have a few components that need some design tweaks so I'll make a plan for component updates.

bartaz commented:
Fix proposed in #1352
